File: unity_tool/photo反転/flip_image.py
import sys
from PIL import Image

def main():
    if len(sys.argv) < 2:
        print("Usage: python flip_image.py <image_file>")
        return

    image_path = sys.argv[1]

    try:
        # 画像を開く
        with Image.open(image_path) as img:
            # 左右反転（水平反転）する
            flipped_img = img.transpose(Image.FLIP_LEFT_RIGHT)

            # 出力ファイル名を生成 (元のファイル名に _flipped を追加)
            parts = image_path.rsplit(".", 1)
            if len(parts) == 2:
                output_path = parts[0] + "_flipped." + parts[1]
            else:
                output_path = image_path + "_flipped"

            # 反転画像を保存
            flipped_img.save(output_path)
            print(f"Flipped image saved as: {output_path}")

    except Exception as e:
        print("Error processing image:", e)

if __name__ == "__main__":
    main()

File: unity_tool/photo反転/FlipImage.cs
using System;
using System.Drawing;
using System.IO;

namespace FlipImage
{
    class Program
    {
        static void Main(string[] args)
        {
            if (args.Length < 1)
            {
                Console.WriteLine("Usage: FlipImage.exe <image_file>");
                return;
            }

            string imagePath = args[0];

            try
            {
                using (Bitmap bmp = new Bitmap(imagePath))
                {
                    // 左右反転：RotateNoneFlipX
                    bmp.RotateFlip(RotateFlipType.RotateNoneFlipX);

                    string outputPath = GenerateOutputFileName(imagePath);
                    bmp.Save(outputPath);
                    Console.WriteLine("Flipped image saved as: " + outputPath);
                }
            }
            catch (Exception ex)
            {
                Console.WriteLine("Error processing image: " + ex.Message);
            }
        }

        private static string GenerateOutputFileName(string imagePath)
        {
            string directory = Path.GetDirectoryName(imagePath);
            string fileName = Path.GetFileNameWithoutExtension(imagePath);
            string extension = Path.GetExtension(imagePath);
            return Path.Combine(directory, fileName + "_flipped" + extension);
        }
    }
}
